### PR TITLE
[Batch] Add CommunicationMode properties to CloudPool

### PR DESCRIPTION
**Pull request by paterasMSFT**
## Description

* Added new properties `CurrentNodeCommunicationMode` (read only) and `TargetCommunicationMode` of type `NodeCommunicationMode` to `PSCloudPool`.
  - Valid values for `NodeCommunicationMode`: Default, Classic, Simplified
  - When the `PSCloudPool` is updated with a new `TargetCommunicationMode` value, the Batch service will attempt to update the pool to the new value the next time the pool is resized down to zero compute nodes and back up.

## Checklist

- [x] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** select appropriate branch. Cmdlets from Autorest.PowerShell should go to [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch. 
* **SHOULD** make the title of PR clear and informative, and in the present imperative tense. 
* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. Add changelog in description section if PR goes into [`generation`](https://github.com/Azure/azure-powershell/tree/generation) branch.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** have approved design review for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) ([_Microsoft internal only_](../blob/main/CONTRIBUTING.md#onboarding)) with following situations
    * Create new module from scratch
    * Create new resource types which are not easy to conform to [Azure PowerShell Design Guidelines](../blob/main/documentation/development-docs/design-guidelines)
    * Create new resource type which name doesn't use module name as prefix
    * Have design question before implementation
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** introduce [breaking changes](../blob/main/documentation/breaking-changes/breaking-changes-definition.md) in Az minor release except preview version.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

**Comment by msJinLei** [comment]
```
2022-12-08T20:52:45.0270073Z   D:\a\_work\1\s\artifacts//StaticAnalysisResults\BreakingChangeIssues.csv Errors
2022-12-08T20:52:45.0271597Z   "Module","ClassName","Target","Severity","ProblemId","Description","Remediation"
2022-12-08T20:52:45.0273653Z   "Az.Batch","Microsoft.Azure.Commands.Batch.DisableBatchAutoScaleCommand","Disable-AzBatchAutoScale","0","3010","The property 'ActionRequired' of type 'Microsoft.Azure.Management.Batch.Models.PrivateLinkServiceConnectionState' has been removed.","Add the property 'ActionRequired' back to type 'Microsoft.Azure.Management.Batch.Models.PrivateLinkServiceConnectionState'.
```

File: src/Batch/Batch/Models/PSPrivateLinkServiceConnectionState.cs [hunk 1]
```diff
@@ -58,7 +58,7 @@ internal static PSPrivateLinkServiceConnectionState CreateFromPrivateLinkService
             return new PSPrivateLinkServiceConnectionState(
                 privateLinkServiceConnectionState.Status,
                 privateLinkServiceConnectionState.Description,
-                privateLinkServiceConnectionState.ActionRequired);
```
Comment (msJinLei) [inline src/Batch/Batch/Models/PSPrivateLinkServiceConnectionState.cs:61]: Breaking changes from `ActionRequired` to `ActionsRequired`

Comment (paterasMSFT) [reply src/Batch/Batch/Models/PSPrivateLinkServiceConnectionState.cs:61]: Good catch. Fixed. Thank you.

**Review by msJinLei — APPROVED** [review]

**Pull request merged**